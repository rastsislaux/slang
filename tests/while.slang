/*
    THIS FILE IS A PART OF SLANG EXAMPLES.
    AUTHOR: Rostislav Lipsky (https://github.com/ungaf)
    DATE: 26.03.2023
    VISIT https://github.com/ungaf/slang AND READ THE LICENSE BEFORE USING.

    Introduced while keyword.
*/

include std.io

const CAP 15

macro main using std.io namespaces

    $number 0 &
    True while // While keyword needs a boolean, if true - while block will be executed, false - skip
        $number @ ++ println!
        $number swap &

        True while  // Whiles can be nested
            "Print once" println!
            False
        do

        $number @ CAP! != // Putting a boolean in the end (true - execute again, false - stop)
    do

end