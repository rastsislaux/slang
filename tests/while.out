1
Print once
2
Print once
3
Print once
4
Print once
5
Print once
6
Print once
7
Print once
8
Print once
9
Print once
10
Print once
11
Print once
12
Print once
13
Print once
14
Print once
15
Print once
